src/frontend/components/NetworkSwitch/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { useCallback, useEffect } from "react";
+import { useSearchParams } from "react-router-dom";
+import { Network, NETWORK_SEARCH_PARAM } from "frontend/constants/settings";
+
+import "./styles.scss";
+
+export const NetworkSwitch = () => {
+  const networkOptions: string[] = Object.values(Network);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const networkParamValue = searchParams.get(NETWORK_SEARCH_PARAM);
+
+  const updateNetworkValue = useCallback(
```

Comment:
:wave: We are using `updateNetworkValue()` in `useEffect` below, so the `useCallback` is needed to make sure we're not re-rendering unnecessarily. Since `updateNetworkValue()` is a function (object in JS world), it will always trigger an update in `useEffect` if it's not wrapped in `useCallback`. Hope that explains a bit.